user_model.ml

```diff
@@ -0,0 +1,82 @@
+module Rng = Mirage_crypto_rng
+
+type token = {
+  token_type : string;
+  access_token : string;
+  expires_in : int; (* After 1 hour (3660 seconds) of inactivity *)
+}
+let clean_string s =
+  (* Remove backslashes and double quotes from the string *)
+  let buffer = Buffer.create (String.length s) in
+  String.iter (fun c ->
+    match c with
+    | '\\' -> ()
+    | '"' -> ()
+    | _ -> Buffer.add_char buffer c
+  ) s;
+  Buffer.contents buffer
+
+
+let token_to_json t : Yojson.Basic.t  =
+  `Assoc [
+    ("token_type", `String t.token_type);
+    ("access_token", `String t.access_token);
+    ("expires_in", `Int t.expires_in)
+  ]
+
+type user = {
+  name : string;
+  email : string;
+  password : string;
+  uuid : string;
+  token : token option;
+}
+
+let user_to_json u : Yojson.Basic.t =
+  `Assoc [
+    ("name", `String u.name);
+    ("email", `String u.email);
+    ("password", `String u.password);
+    ("uuid", `String u.uuid);
+    ("token", match u.token with
+              | Some t -> token_to_json t
+              | None -> `Null)
+  ]
+
+let encrypt_password password uuid =
+  let enc = Mirage_crypto.Hash.SHA256.digest
+    (Cstruct.of_string (uuid ^ "-" ^ password)) in
+  match Base64.encode (Cstruct.to_string enc) with
+  | Error (`Msg s) ->
+      Logs.warn (fun m ->
+          m "Failed to generate user token for: %s" s);
+      None
+  | Ok pass -> Some pass
+
+let generate_uuid ~email =
+  let email_length = String.length email in
+  Uuidm.v4_gen (Random.State.make [| email_length |]) ()
```

Comment:
```suggestion
let generate_uuid () =
  let data = Rng.generate 16 in
  Uuidm.v4 (Cstruct.to_bytes data)
```

Reply:
For what it is worth: the documentation of `Uuidm.v4_gen` mentions:     {b Warning.} Sequences of UUIDs generated using {!Stdlib.Random} are suitably random but {e predictable} by an observer. If that is an issue for you, use {!v4} with random bytes generated by a CSPRNG. *)

This may be a bit cryptic, but we have the `Rng`, which is a "Cryptographically Secure Random Number Generator (CSPRNG)". And we just generate 16 random bytes, and treat those as a UUID :)